agent/rpc_from.go

```diff
@@ -38,8 +28,41 @@ func (a *orbAgent) handleGroupMembership(rpc fleet.GroupMembershipRPCPayload) {
 	}
 }
 
-func (a *orbAgent) handleAgentPolicies(rpc []fleet.AgentPolicyRPCPayload) {
-
+func (a *orbAgent) handleAgentPolicies(rpc []fleet.AgentPolicyRPCPayload, fullList bool) {
+	if fullList {
+		policies, err := a.policyManager.GetRepo().GetAll()
+		if err != nil {
+			a.logger.Error("failed to retrieve policies on handle subscriptions")
+		}
+		// Compare the list of policies the check which one should be removed
+		// After a editing a policy
+		policyRemove := map[string]bool{}
+		for _, p := range policies {
```

Comment:
this is O(n2) in complexity, not a problem for our small policy counts right now but would be if there were hundreds of policies. it might make sense to make a hash table of rpc.payload.ID => bool before the `range policies` loop and use that instead of the embedded loop